testsrc/main/mondrian/rolap/aggmatcher/AggMeasureFactCountTest.java

```diff
@@ -0,0 +1,288 @@
+                + "</AggName>\n";
+
+        String schema = getAggSchema(agg);
+
+        verifySameAggAndNot(QUERY, schema);
+    }
+
+    public void testAggDivideByZero() {
+        String agg = ""
+                + "<AggExclude name=\"agg_c_6_fact_csv_2016\" />"
+                + "<AggName name=\"agg_csv_divide_by_zero\">\n"
+                + "    <AggFactCount column=\"FACT_COUNT\"/>\n"
+                + "    <AggMeasureFactCount column=\"store_sales_fact_count\" factColumn=\"store_sales\" />\n"
+                + "    <AggMeasureFactCount column=\"store_cost_fact_count\" factColumn=\"store_cost\" />\n"
+                + "    <AggMeasureFactCount column=\"unit_sales_fact_count\" factColumn=\"unit_sales\" />\n"
+                + "    <AggMeasure name=\"[Measures].[Unit Sales]\" column=\"UNIT_SALES\" />\n"
+                + "    <AggMeasure name=\"[Measures].[Store Cost]\" column=\"STORE_COST\" />\n"
+                + "    <AggMeasure name=\"[Measures].[Store Sales]\" column=\"STORE_SALES\" />\n"
+                + "    <AggLevel name=\"[Time].[Year]\" column=\"the_year\" />\n"
+                + "    <AggLevel name=\"[Time].[Quarter]\" column=\"quarter\" />\n"
+                + "    <AggLevel name=\"[Time].[Month]\" column=\"month_of_year\" />\n"
+                + "</AggName>\n";
+
+        String result = ""
+                + "Axis #0:\n"
+                + "{}\n"
+                + "Axis #1:\n"
+                + "{[Time].[1997].[Q1]}\n"
+                + "{[Time].[1997].[Q2]}\n"
+                + "{[Time].[1997].[Q3]}\n"
+                + "{[Time].[1997].[Q4]}\n"
+                + "Axis #2:\n"
+                + "{[Measures].[Store Sales]}\n"
+                + "{[Measures].[Store Cost]}\n"
+                + "{[Measures].[Unit Sales]}\n"
+                + "Row #0: \n"
+                + "Row #0: 1.00\n"
+                + "Row #0: 1.00\n"
+                + "Row #0: 1.00\n"
+                + "Row #1: 2.00\n"
+                + "Row #1: 2.00\n"
+                + "Row #1: 2.00\n"
+                + "Row #1: 2.00\n"
+                + "Row #2: 3\n"
+                + "Row #2: 3\n"
+                + "Row #2: 3\n"
+                + "Row #2: 3\n";
+
+        String schema = getAggSchema(agg);
+        TestContext testContext = getTestContext().withSchema(schema);
+        testContext.assertQueryReturns(QUERY, result);
+    }
+
+    public void testAggPattern() {
+        String agg = ""
+                + "<AggPattern pattern=\"agg_c_6_fact_csv_2016\">\n"
+                + "    <AggFactCount column=\"FACT_COUNT\"/>\n"
+                + "    <AggMeasureFactCount column=\"store_sales_fact_count\" factColumn=\"store_sales\" />\n"
+                + "    <AggMeasureFactCount column=\"store_cost_fact_count\" factColumn=\"store_cost\" />\n"
+                + "    <AggMeasureFactCount column=\"unit_sales_fact_count\" factColumn=\"unit_sales\" />\n"
```

Comment:
Is the factColumn value supposed to be case sensitive? It is different here as compared to the definition within the AggMeasure element. What happens if the DB does not match those? Please try it / test it.